xbmc/video/videosync/VideoSyncCocoa.cpp

```diff
@@ -33,6 +33,28 @@
 #if defined(TARGET_DARWIN_OSX)
 #include <QuartzCore/CVDisplayLink.h>
 #include "osx/CocoaInterface.h"
+#endif
+
+void CVideoSyncCocoa::OnLostDevice()
+{
+  if (!m_displayLost)
+  {
+    m_displayLost = true;
+    m_lostEvent.Wait();
```

Comment:
sry my comment was ment to go here - i don't see how onresetdevice would be able to NOT terminate Run()